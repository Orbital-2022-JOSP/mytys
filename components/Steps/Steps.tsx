import { GenericContainer } from '../GenericContainer/GenericContainer';

export const Steps: React.FC = () => {
    return (
        <GenericContainer>
            <p className="mb-2 text-3xl gap-0 font-bold text-light-blue-500">Our Simple Learning Process</p>
            <div className="grid gap-6 row-gap-10 lg:grid-cols-2">
                <div className="lg:py-6 lg:pr-16">
                    <div className="flex">
                        <div className="flex flex-col items-center mr-4">
                            <div>
                                <div className="flex items-center justify-center w-10 h-10 border rounded-full">
                                    <svg
                                        className="w-4 text-light-blue-500"
                                        stroke="currentColor"
                                        strokeWidth="2"
                                        strokeLinecap="round"
                                        strokeLinejoin="round"
                                        viewBox="0 0 24 24"
                                    >
                                        <line
                                            fill="none"
                                            strokeMiterlimit="10"
                                            x1="12"
                                            y1="2"
                                            x2="12"
                                            y2="22"
                                        />
                                        <polyline
                                            fill="none"
                                            strokeMiterlimit="10"
                                            points="19,15 12,22 5,15"
                                        />
                                    </svg>
                                </div>
                            </div>
                            <div className="w-px h-full bg-gray-300" />
                        </div>
                        <div className="pt-1 pb-8">
                            <p className="mb-2 text-lg font-bold text-light-blue-500">Log In</p>
                            <p className="text-gray-700">
                                Log In to your account and access the wide array of curated practices you can choose from
                            </p>
                        </div>
                    </div>
                    <div className="flex">
                        <div className="flex flex-col items-center mr-4">
                            <div>
                                <div className="flex items-center justify-center w-10 h-10 border rounded-full">
                                    <svg
                                        className="w-4 text-light-blue-500"
                                        stroke="currentColor"
                                        strokeWidth="2"
                                        strokeLinecap="round"
                                        strokeLinejoin="round"
                                        viewBox="0 0 24 24"
                                    >
                                        <line
                                            fill="none"
                                            strokeMiterlimit="10"
                                            x1="12"
                                            y1="2"
                                            x2="12"
                                            y2="22"
                                        />
                                        <polyline
                                            fill="none"
                                            strokeMiterlimit="10"
                                            points="19,15 12,22 5,15"
                                        />
                                    </svg>
                                </div>
                            </div>
                            <div className="w-px h-full bg-gray-300" />
                        </div>
                        <div className="pt-1 pb-8">
                            <p className="mb-2 text-lg font-bold text-light-blue-500">Participate and practice</p>
                            <p className="text-gray-700">
                                Participate in the challenges we have on the platform, and practice your way through exercises offered on our exclusive gamified MYTYS platform
                            </p>
                        </div>
                    </div>
                    <div className="flex">
                        <div className="flex flex-col items-center mr-4">
                            <div>
                                <div className="flex items-center justify-center w-10 h-10 border rounded-full">
                                    <svg
                                        className="w-4 text-light-blue-500"
                                        stroke="currentColor"
                                        strokeWidth="2"
                                        strokeLinecap="round"
                                        strokeLinejoin="round"
                                        viewBox="0 0 24 24"
                                    >
                                        <line
                                            fill="none"
                                            strokeMiterlimit="10"
                                            x1="12"
                                            y1="2"
                                            x2="12"
                                            y2="22"
                                        />
                                        <polyline
                                            fill="none"
                                            strokeMiterlimit="10"
                                            points="19,15 12,22 5,15"
                                        />
                                    </svg>
                                </div>
                            </div>
                            <div className="w-px h-full bg-gray-300" />
                        </div>
                        <div className="pt-1 pb-8">
                            <p className="mb-2 text-lg font-bold text-light-blue-500">Get Rewarded</p>
                            <p className="text-gray-700">
                                Receive points based on your practice scores and your particpation, having a chance to haved your name displayed on the leaderboards
                            </p>
                        </div>
                    </div>
                    <div className="flex">
                        <div className="flex flex-col items-center mr-4">
                            <div>
                                <div className="flex items-center justify-center w-10 h-10 border rounded-full">
                                    <svg
                                        className="w-4 text-light-blue-500"
                                        stroke="currentColor"
                                        strokeWidth="2"
                                        strokeLinecap="round"
                                        strokeLinejoin="round"
                                        viewBox="0 0 24 24"
                                    >
                                        <line
                                            fill="none"
                                            strokeMiterlimit="10"
                                            x1="12"
                                            y1="2"
                                            x2="12"
                                            y2="22"
                                        />
                                        <polyline
                                            fill="none"
                                            strokeMiterlimit="10"
                                            points="19,15 12,22 5,15"
                                        />
                                    </svg>
                                </div>
                            </div>
                            <div className="w-px h-full bg-gray-300" />
                        </div>
                        <div className="pt-1 pb-8">
                            <p className="mb-2 text-lg font-bold text-light-blue-500">Review your mistakes and learn from answers</p>
                            <p className="text-gray-700">
                                Upon completion of marking by your assigned tutor, you can review your mistakes and learn from them
                            </p>
                        </div>
                    </div>
                    <div className="flex">
                        <div className="flex flex-col items-center mr-4">
                            <div>
                                <div className="flex items-center justify-center w-10 h-10 border rounded-full">
                                    <svg
                                        className="w-6 text-light-blue-500"
                                        stroke="currentColor"
                                        viewBox="0 0 24 24"
                                    >
                                        <polyline
                                            fill="none"
                                            strokeWidth="2"
                                            strokeLinecap="round"
                                            strokeLinejoin="round"
                                            strokeMiterlimit="10"
                                            points="6,12 10,16 18,8"
                                        />
                                    </svg>
                                </div>
                            </div>
                        </div>
                        <div className="pt-1">
                            <p className="mb-2 text-lg font-bold text-light-blue-500">As simple as it gets!</p>
                            <p className="text-gray-700" />
                        </div>
                    </div>
                </div>
                <div className="relative">
                    <img
                        className="inset-0 object-cover object-bottom w-full rounded shadow-lg h-96 lg:absolute lg:h-full"
                        src="https://images.pexels.com/photos/3184287/pexels-photo-3184287.jpeg?auto=compress&amp;cs=tinysrgb&amp;dpr=2&amp;h=750&amp;w=1260"
                        alt=""
                    />
                </div>
            </div>
        </GenericContainer>
    );
};