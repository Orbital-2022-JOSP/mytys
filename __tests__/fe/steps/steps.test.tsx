import { render, screen } from '@testing-library/react';
import { Steps } from '../../../components/Steps/Steps';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Steps Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<Steps />);

            expect(container).toMatchSnapshot()
            expect(screen.getByText("Our Simple Learning Process"))
                .toBeInTheDocument();
        })
})

