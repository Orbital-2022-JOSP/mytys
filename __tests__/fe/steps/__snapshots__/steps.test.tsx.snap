// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Steps Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <p
      class="mb-2 text-3xl gap-0 font-bold text-light-blue-500"
    >
      Our Simple Learning Process
    </p>
    <div
      class="grid gap-6 row-gap-10 lg:grid-cols-2"
    >
      <div
        class="lg:py-6 lg:pr-16"
      >
        <div
          class="flex"
        >
          <div
            class="flex flex-col items-center mr-4"
          >
            <div>
              <div
                class="flex items-center justify-center w-10 h-10 border rounded-full"
              >
                <svg
                  class="w-4 text-light-blue-500"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                  <line
                    fill="none"
                    stroke-miterlimit="10"
                    x1="12"
                    x2="12"
                    y1="2"
                    y2="22"
                  />
                  <polyline
                    fill="none"
                    points="19,15 12,22 5,15"
                    stroke-miterlimit="10"
                  />
                </svg>
              </div>
            </div>
            <div
              class="w-px h-full bg-gray-300"
            />
          </div>
          <div
            class="pt-1 pb-8"
          >
            <p
              class="mb-2 text-lg font-bold text-light-blue-500"
            >
              Log In
            </p>
            <p
              class="text-gray-700"
            >
              Log In to your account and access the wide array of curated practices you can choose from
            </p>
          </div>
        </div>
        <div
          class="flex"
        >
          <div
            class="flex flex-col items-center mr-4"
          >
            <div>
              <div
                class="flex items-center justify-center w-10 h-10 border rounded-full"
              >
                <svg
                  class="w-4 text-light-blue-500"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                  <line
                    fill="none"
                    stroke-miterlimit="10"
                    x1="12"
                    x2="12"
                    y1="2"
                    y2="22"
                  />
                  <polyline
                    fill="none"
                    points="19,15 12,22 5,15"
                    stroke-miterlimit="10"
                  />
                </svg>
              </div>
            </div>
            <div
              class="w-px h-full bg-gray-300"
            />
          </div>
          <div
            class="pt-1 pb-8"
          >
            <p
              class="mb-2 text-lg font-bold text-light-blue-500"
            >
              Participate and practice
            </p>
            <p
              class="text-gray-700"
            >
              Participate in the challenges we have on the platform, and practice your way through exercises offered on our exclusive gamified MYTYS platform
            </p>
          </div>
        </div>
        <div
          class="flex"
        >
          <div
            class="flex flex-col items-center mr-4"
          >
            <div>
              <div
                class="flex items-center justify-center w-10 h-10 border rounded-full"
              >
                <svg
                  class="w-4 text-light-blue-500"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                  <line
                    fill="none"
                    stroke-miterlimit="10"
                    x1="12"
                    x2="12"
                    y1="2"
                    y2="22"
                  />
                  <polyline
                    fill="none"
                    points="19,15 12,22 5,15"
                    stroke-miterlimit="10"
                  />
                </svg>
              </div>
            </div>
            <div
              class="w-px h-full bg-gray-300"
            />
          </div>
          <div
            class="pt-1 pb-8"
          >
            <p
              class="mb-2 text-lg font-bold text-light-blue-500"
            >
              Get Rewarded
            </p>
            <p
              class="text-gray-700"
            >
              Receive points based on your practice scores and your particpation, having a chance to haved your name displayed on the leaderboards
            </p>
          </div>
        </div>
        <div
          class="flex"
        >
          <div
            class="flex flex-col items-center mr-4"
          >
            <div>
              <div
                class="flex items-center justify-center w-10 h-10 border rounded-full"
              >
                <svg
                  class="w-4 text-light-blue-500"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                  <line
                    fill="none"
                    stroke-miterlimit="10"
                    x1="12"
                    x2="12"
                    y1="2"
                    y2="22"
                  />
                  <polyline
                    fill="none"
                    points="19,15 12,22 5,15"
                    stroke-miterlimit="10"
                  />
                </svg>
              </div>
            </div>
            <div
              class="w-px h-full bg-gray-300"
            />
          </div>
          <div
            class="pt-1 pb-8"
          >
            <p
              class="mb-2 text-lg font-bold text-light-blue-500"
            >
              Review your mistakes and learn from answers
            </p>
            <p
              class="text-gray-700"
            >
              Upon completion of marking by your assigned tutor, you can review your mistakes and learn from them
            </p>
          </div>
        </div>
        <div
          class="flex"
        >
          <div
            class="flex flex-col items-center mr-4"
          >
            <div>
              <div
                class="flex items-center justify-center w-10 h-10 border rounded-full"
              >
                <svg
                  class="w-6 text-light-blue-500"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <polyline
                    fill="none"
                    points="6,12 10,16 18,8"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="pt-1"
          >
            <p
              class="mb-2 text-lg font-bold text-light-blue-500"
            >
              As simple as it gets!
            </p>
            <p
              class="text-gray-700"
            />
          </div>
        </div>
      </div>
      <div
        class="relative"
      >
        <img
          alt=""
          class="inset-0 object-cover object-bottom w-full rounded shadow-lg h-96 lg:absolute lg:h-full"
          src="https://images.pexels.com/photos/3184287/pexels-photo-3184287.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
        />
      </div>
    </div>
  </div>
</div>
`;
